src/main/scala/firrtl/passes/RemoveCHIRRTL.scala

```diff
@@ -118,7 +119,7 @@ object RemoveCHIRRTL extends Pass {
         set_write(rws, "wdata", "wmask")
       val mem = DefMemory(s.info, s.name, s.tpe, s.size, 1, if (s.seq) 1 else 0,
                   rds map (_.name), wrs map (_.name), rws map (_.name))
-      Block(mem +: stmts)
+      if (stmts.nonEmpty) Block(mem +: stmts) else EmptyStmt
```

Comment:
If a memory has no ports, why should it exist? If deleting the memory will make it so that other passes have no reason to be sad... isn't that the right decision?

I guess it'd still be problematic if someone wrote FIRRTL but not CHIRRTL and then passed it through some intermediate lowering compiler... You'd need some kind of clean up pass before every pass that can break.